checks.d/elastic.py

```diff
@@ -322,16 +322,20 @@ def check(self, instance):
         health_url, nodes_url, stats_url, pshard_stats_url, pending_tasks_url, stats_metrics, \
             pshard_stats_metrics = self._define_params(version, config.cluster_stats)
 
+        # Load stats data.
+        # This must happen before other URL processing as the cluster name
+        # is retreived here, and added to the tag list.
+
+        stats_url = urlparse.urljoin(config.url, stats_url)
+        stats_data = self._get_data(stats_url, config)
+        self._process_stats_data(nodes_url, stats_data, stats_metrics, config)
```

Comment:
I'm not a huge fan either of modifying `config` after its creation, but I don't think there is another way to do this.
Maybe we could put the config update in the `check` function ?
Something like:
```python
stats_data = self._get_data(stats_url, config)
config.tags.append("cluster_name:{}".format(stats_data['cluster_name']))
self._process_stats_data(nodes_url, stats_data, stats_metrics, config)
```
This way it would be more obvious that it's being changed. What do you think ?